### PR TITLE
Describe instances

### PR DESCRIPTION
**Pull request by clementbiron**

### CODE REVIEW

File: src/pages/homepage.tsx
Comment (MattiSG) [inline src/pages/homepage.tsx:287]: Let the browser help users with autofill and validation 🙂 

```suggestion
                    type="email"
```

File: src/pages/homepage.tsx
Comment (MattiSG) [inline src/pages/homepage.tsx:293]: Why both?

```suggestion
                    required
```

**Review by MattiSG — CHANGES_REQUESTED** [review]
I had over 50% weight gain running images through ImageOptim. I will commit those. Please remember to optimise images before committing them, in order to minimise ecological costs 🙂
I will also change some of the copy.

The Reset and HestiaLabs logos are blurry. If you cannot find higher quality ones on the internet, please ask the partners directly 🙂

**Comment by MattiSG** [comment]
### CTAs are not typeset on a grid

The distance between the title and the body text should be the same as the distance between the body text and the button.

Screen Shot 2022-09-06 at 17 32 51

### Instances descriptions are duplicated with their metadata

Screen Shot 2022-09-06 at 17 33 19

### Instances descriptions are inconsistent with their metadata

See “languages” on Contrib.

Screen Shot 2022-09-06 at 17 33 04


As it is, the description + metadata of instances does not work. We can have a synchronous call on it if useful 🙂

**Comment by MattiSG** [comment]
“Volunteer contributors” looks rather small compared to other contributors, could we maybe have a nice image for them too? 🙂 

Screen Shot 2022-09-06 at 17 42 49

File: public/instances.json
Comment (clementbiron) [reply public/instances.json]: Fixed with https://github.com/ambanum/OpenTermsArchive.org/pull/161/commits/f533a64d40a71ff825235112714bf1f3417dcfab

File: src/translations/en/instances.json
Comment (clementbiron) [reply src/translations/en/instances.json]: We absolutly don't need both, that's a configuration mistake fixed by https://github.com/ambanum/OpenTermsArchive.org/pull/161/commits/e03010673c18b96519e1cb24494a109be9f31540

File: src/pages/homepage.tsx
Comment (clementbiron) [reply src/pages/homepage.tsx]: Done with https://github.com/ambanum/OpenTermsArchive.org/pull/161/commits/4b0cea8844aa47ec5c7512fe484244d4080d5d33

File: src/pages/homepage.tsx
Comment (clementbiron) [reply src/pages/homepage.tsx]: Done with https://github.com/ambanum/OpenTermsArchive.org/pull/161/commits/4b0cea8844aa47ec5c7512fe484244d4080d5d33

File: src/pages/homepage.tsx
Comment (clementbiron) [reply src/pages/homepage.tsx]: Done with https://github.com/ambanum/OpenTermsArchive.org/pull/161/commits/1830511824d4a8074ea666fe211e3a0efda7128c

File: src/pages/homepage.tsx
Comment (clementbiron) [reply src/pages/homepage.tsx:288]: This ID is created by SendInBlue (I can't change it) and i think that HTML attributes values are [case-sensitive](https://www.w3.org/TR/2010/WD-html-markup-20101019/documents.html#case-insensitivity).

File: src/translations/fr/instances.json
Comment (clementbiron) [reply src/translations/fr/instances.json]: Improve by https://github.com/ambanum/OpenTermsArchive.org/pull/161/commits/99308a46053e06d4293de69532ca6176b6d0f52b

From:
image

To:
image

File: src/pages/homepage.tsx
Comment (MattiSG) [reply src/pages/homepage.tsx:307]: User-provided values are indeed case-sensitive. However, unless some scripts to magic, the value that will be passed through the form to SendInBlue is provided by the `name` attribute, not the `id` which is only valid in the DOM of the current page 🙂

File: src/translations/fr/homepage.json
Comment (MattiSG) [inline src/translations/fr/homepage.json:18]:
```suggestion
  "how.bloc4.desc": "Les versions enregistrées sont publiées dans des jeux de données faciles à analyser.",
```

File: src/modules/I18n/i18next-scanner.config.js
Comment (MattiSG) [inline src/modules/I18n/i18next-scanner.config.js:44]: > Why change this? It seems quite positive to me that we don't maintain unused translations :)

File: src/pages/homepage.tsx
Comment (MattiSG) [inline src/pages/homepage.tsx:106]: This is ambiguous with the `slug` of the URLs, and not very clear in the following code.

```suggestion
              const instanceId = kebabCase(instance.name);
```

File: public/instances.json
Comment (MattiSG) [inline public/instances.json:4]: > Why not simply use the `name` as the key and list them all at root level? This would spare you from having to navigate the data through `instancesData.instances`, for example 🙂

File: public/instances.json
Comment (MattiSG) [inline public/instances.json:5]: > Since we already define dynamically the `id` (and thus all links) of the instance from its `kebabCase(name)`, I wouldn't be shocked if we also assumed the `image` is always at `/images/instances/${id}.png` 😉

File: public/instances.json
Comment (MattiSG) [inline public/instances.json:7]:
```suggestion
      "jurisdictions": ["EU", "US"],
```

File: public/instances.json
Comment (MattiSG) [inline public/instances.json:19]: Please use consistent casing between the two codes 🙂

**Review by MattiSG — CHANGES_REQUESTED** [review]
### Illustrations

- For the `Contrib` instance illustration, could we maybe have a globe / web icon instead of documents with contributors in them? For all other instances, we indeed provide a representation of the target scope, so I would find it more consistent 🙂 
- As already reported earlier, some partner logos are blurry.

### Instance metadata presentation

The new presentation is much nicer than a list with discs! However, the icons are not currently semantically satisfactory 🙂 
Screen Shot 2022-09-07 at 18 28 47

- Services should be an “app” or “service” icon; worst case “gear”.
- Language should be a language (not nationality) icon, such as “translation” (half roman script half ideogram), “speech bubble” or “letter”.
- Jurisdiction should be a “map” or “flag” or “institution” icon.

### Instance exploration CTAs

I feel this gives many first-level CTAs. Do we really want to push users to open every single one of those? 🙂 

Screen Shot 2022-09-07 at 18 28 53

I'm in favour of a series of tertiary CTAs (links like “download dataset” and “subscribe over email”), if possible with leading icons.

### Instance descriptions

Switching to mostly metadata presentation indeed fixes the problem of duplicated information… in most cases. I see that `France-election` still has its  `france-elections.desc`. This makes for a special case, and shows that we need some free-form description.
I suggest that we add an `industries` field in the instances JSON data, which would be an object with language code as keys, translating the industries scope.

For example, for `france-elections`, `industries` would be:

```json
"industries": {
  "en": "most used social media in France that could have a systemic impact on the 2022 elections",
  "fr": "médias sociaux pouvant influencer les élections présidentielles et législatives françaises 2022"
}
```

and `dating` would have:

```json
"industries": {
  "en": "online dating",
  "fr": "rencontre en ligne"
}
```

etc. WDYT? 🙂 Happy to have a call if useful.

File: src/modules/Common/components/Card.tsx
Comment (martinratinaud) [inline src/modules/Common/components/Card.tsx:28]: not blocking but `authorIcon` let me think I can pass an icon of my own. Would have used `withAuthorIcon`

File: src/modules/Common/components/Card.tsx
Comment (martinratinaud) [inline src/modules/Common/components/Card.tsx:47]: Classname can be used with objects

```
classNames(
  s.card,
  {
    link: s.card__isLink,
    center: s.card__center,
  ...
  }
```

Although the notation might seem a bit weird, in the first place, it's actually easier to understand IMO.

File: src/modules/Common/components/Card.tsx
Comment (martinratinaud) [inline src/modules/Common/components/Card.tsx:64]: `authorCenter` is already a boolean `[s.card_author__isCenter]: authorCenter,`

File: src/modules/Common/components/Card.tsx
Comment (martinratinaud) [inline src/modules/Common/components/Card.tsx:69]: color could be set in card_author_icon css class to use the css vars

File: src/modules/Common/components/Card.tsx
Comment (martinratinaud) [inline src/modules/Common/components/Card.tsx:75]: is it normal that this link is empty?

File: src/modules/Common/components/CardList.tsx
Comment (martinratinaud) [inline src/modules/Common/components/CardList.tsx:35]: already booleans, no need for `=== true`

File: src/modules/Common/components/CardTableItem.tsx
Comment (martinratinaud) [inline src/modules/Common/components/CardTableItem.tsx:30]: no need for `classNames` here

File: src/modules/Common/components/TextContent.tsx
Comment (martinratinaud) [inline src/modules/Common/components/TextContent.tsx:8]: Not blocking, again, I intend to put a number here, based on the wording

File: src/modules/Common/components/TextContent.tsx
Comment (martinratinaud) [inline src/modules/Common/components/TextContent.tsx]: Use `!marginTop` in case undefined is passed

```
classNames(
        s.textContent,
        {
          [s.textContent__marginTopLarge]: marginTopLarge,
          [s.textContent__hasNoMarginTop]: !marginTop,
        },
        className
      )
```

File: src/modules/I18n/i18next-scanner.config.js
Comment (martinratinaud) [reply src/modules/I18n/i18next-scanner.config.js:44]: I think it's because there are dynamic translation keys that would be removed because of this.

File: src/pages/homepage.tsx
Comment (martinratinaud) [inline src/pages/homepage.tsx:190]: This could be in a translation string

File: src/modules/Common/components/Instances.tsx
Comment (martinratinaud) [inline src/modules/Common/components/Instances.tsx:32]: Not blocking

Here we should 
- have the same size for all maintainers and contributors assets
- use `next/image` with fixed size

File: src/modules/Common/components/Instances.tsx
Comment (martinratinaud) [inline src/modules/Common/components/Instances.tsx]: if you have the kebabcase slug, you can remove `uniqueId` and do `key={`instance_${slug}`}`

File: src/pages/homepage.tsx
Comment (martinratinaud) [inline src/pages/homepage.tsx:134]: For the record, you don't need the `={true}`

**Review by martinratinaud — APPROVED** [review]
Good job, it looks good.

I fine tuned some components structure, I advise you @clementbiron to take a look at my commits.

**Review by martinratinaud — CHANGES_REQUESTED** [review]
I also rebased on main

File: src/modules/Common/components/Card.tsx
Comment (MattiSG) [reply src/modules/Common/components/Card.tsx:28]: +1, or `hasAuthorIcon`.

**Comment by MattiSG** [comment]
> I also rebased on main

Thanks for the work that entailed! …but can we please stop force-pushing as long as reviews are ongoing? 😢😉

**Comment by MattiSG** [comment]
> Lighten size of images with ImageOptim

I'm surprised @martinratinaud, could you decrease the sizes further? I tried to run it and had only very marginal gains (-0,1%) on two or three after I had done it once. If you found a way to go further, I'd be curious to learn about your ImageOptim config! 😊

File: public/instances.json
Comment (MattiSG) [reply public/instances.json:19]: Thanks for objecting @martinratinaud! 👍 Thanks to you I learned today that:

- “ISO 639-2 recommends use of the language codes in lower case (but they should be considered case-insensitive and are unique codes regardless of case)”, according to the [US Library of Congress](https://www.loc.gov/standards/iso639-2/faq.html#21). The standard itself is not available for free, so I cannot double-check.
- ISO 3166-1:2020 specifies that “code element shall consist only of […] LATIN CAPITAL LETTER A through LATIN CAPITAL LETTER Z […]”, according to the standard itself (or rather the [leaked version](https://cdn.standards.iteh.ai/samples/72482/cbb6318e772a4f2284bdb0bd51f29017/ISO-3166-1-2020.pdf) since, once more, the standard is not freely accessible).

See also https://github.com/stefangabos/world_countries/issues/50.

I definitely consider jurisdictions as “countries”.

**Comment by martinratinaud** [comment]
> > I also rebased on main
> 
> Thanks for the work that entailed! …but can we please stop force-pushing as long as reviews are ongoing? 😢😉

Hum
How can you rebase without force pushing? I figured what you asked was only to only create new commits (which I did) so I'm not sure what you're asking.

Thanks

**Comment by martinratinaud** [comment]
> > > I also rebased on main
> > 
> > 
> > Thanks for the work that entailed! …but can we please stop force-pushing as long as reviews are ongoing? 😢😉
> 
> Hum How can you rebase without force pushing? I figured what you asked was only to only create new commits (which I did) so I'm not sure what you're asking.
> 
> Thanks

Some of them were only 0,1 but some other more than 50% (I think on favicons). So no I'm sorry I don't have any magic config :-)

**Comment by MattiSG** [comment]
> How can you rebase without force pushing?

Exactly, you can't :wink:

What [we had agreed on](https://github.com/ambanum/OpenTermsArchive/wiki/Guide-board#review-after-force-push) is to not force push as long as a review is ongoing. We just wait until approval to rebase 🙂

**Comment by martinratinaud** [comment]
Well ok sorry then, will not rebase before that next time

File: src/pages/homepage.tsx
Comment (clementbiron) [reply src/pages/homepage.tsx]: Fixed with https://github.com/ambanum/OpenTermsArchive.org/pull/161/commits/3c8d52d473cd2a2bb49f58c348f055c57aa55a80

File: src/modules/I18n/i18next-scanner.config.js
Comment (clementbiron) [reply src/modules/I18n/i18next-scanner.config.js:44]: As Martin said, with the use of dynamc translation keys like `t('instances:${slug}.desc')` we had to pass `removeUnusedKeys` to `false`.

File: src/pages/homepage.tsx
Comment (clementbiron) [reply src/pages/homepage.tsx]: Fixed with https://github.com/ambanum/OpenTermsArchive.org/pull/161/commits/bb493c0b20636a957e5f69f2e2bbc164b8912862

File: public/instances.json
Comment (clementbiron) [reply public/instances.json]: Done with https://github.com/ambanum/OpenTermsArchive.org/pull/161/commits/e19159902f322d612c392ae1eea25f4f0c4dfa08

File: public/instances.json
Comment (clementbiron) [reply public/instances.json]: https://github.com/ambanum/OpenTermsArchive.org/pull/161/commits/e19159902f322d612c392ae1eea25f4f0c4dfa08

File: src/modules/Common/components/Card.tsx
Comment (clementbiron) [reply src/modules/Common/components/Card.tsx:49]: https://github.com/ambanum/OpenTermsArchive.org/pull/161/commits/d365e6ea8b61443d06072efb16209040bd3a557a

File: src/modules/Common/components/Card.tsx
Comment (clementbiron) [reply src/modules/Common/components/Card.tsx]: https://github.com/ambanum/OpenTermsArchive.org/pull/161/commits/dbcd2ac2735d150a6e5718f10bb4ebba22c7bddd

File: src/pages/homepage.tsx
Comment (clementbiron) [reply src/pages/homepage.tsx]: yes of course https://github.com/ambanum/OpenTermsArchive.org/pull/161/commits/c21ee552277322a83b287d20964f2a8ebe951982 !

File: src/modules/Common/components/Instances.tsx
Comment (clementbiron) [reply src/modules/Common/components/Instances.tsx:52]: It's a pain to have the same image size for all logos or to have to declare their size in the `Instances.json` file to use `next/image`.

Let's talk about it.

File: src/modules/Common/components/Instances.tsx
Comment (clementbiron) [reply src/modules/Common/components/Instances.tsx]: `key={'instance_${instanceId}'}` ;)

File: src/pages/homepage.tsx
Comment (clementbiron) [reply src/pages/homepage.tsx:135]: `center` Card prop is false by default 👀 

```typescript
const Card: React.FC<CardProps> = ({
  children,
  center = false,
  ...props
})
```

**Comment by clementbiron** [comment]
Thank you for your reviews 💋 

@MattiSG I took your feedback into account and did my best when I couldn't implement exactly what you suggested. 

I hope this is enough to validate this RP  😓

File: src/modules/Common/components/Card.tsx
Comment (martinratinaud) [reply src/modules/Common/components/Card.tsx:77]: Yes it is as it's in absolute position over the whole card

File: src/pages/homepage.tsx
Comment (martinratinaud) [reply src/pages/homepage.tsx:135]: I mean, you can do
```
author={t('homepage:reuses.tosdr.author')}
link="https://tosdr.org"
center
```

**Review by martinratinaud — APPROVED** [review]

**Review by martinratinaud — APPROVED** [review]

**Comment by martinratinaud** [comment]
I merge despite @MattiSG's change requested as I think everything has been addressed accordingly.

**Pull request merged**
